sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -363,6 +401,8 @@ RT::PiProgram ProgramManager::loadProgram(OSModuleHandle M,
     F.close();
   }
   // Load the selected image
+  if (!is_compiler_available(Context, Format))
+    throw feature_not_supported("Online compilation is not supported by this device");
```

Comment:
by this device -> in this context

Reply:
I was trying to use the same message as the other throws in the file, but I will change it